### DISCUSSION TITLE
Test Angular data binding with cypress

### DISCUSSION THREAD

**Discussion by Atlas1225** [cypress-io/cypress#24835] [Questions and Help] [answered]
I've just started testing my Angular App and I need to check if some bound attributes contain what I want. Let's say I want to check this:
<mat-table data-testid="table" [attr_t]="mydata">.
cy.byTestId('table').invoke('attr','attr_t').should('eq', mydata) doesn't work for data bound attributes.

**Comment by portyu9** [comment, marked as answer]
Hi @Atlas1225 

`[attr_t]="mydata"` is an Angular **property/input binding**, not an HTML attribute binding. Because of that, Cypress/jQuery's `attr()` is not the right thing to read.

If you actually want `attr_t` to exist as an HTML attribute, bind it with Angular's `attr.` syntax:

```html
<mat-table
  data-testid="table"
  [attr.attr_t]="mydata">
</mat-table>
```

Then assert it in Cypress:

```js
cy.byTestId('table')
  .should('have.attr', 'attr_t', String(mydata))
```

HTML attribute values are strings, so if `mydata` is not already a string, compare against its serialized/string value.

If `attr_t` is instead meant to be an Angular component input/property, it may never appear as a DOM attribute. In that case I would test the rendered behavior produced by that input rather than trying to read it with `.attr()`.

Please mark this answer as accepted if it helped, thank you.
